File: python/spdk/rpc/keyring.py
# SPDX-License-Identifier: BSD-3-Clause
# Copyright (c) 2024 Intel Corporation.  All rights reserved.

from spdk.rpc.helpers import deprecated_method


@deprecated_method
def keyring_file_add_key(client, name, path):
    return client.call('keyring_file_add_key', {'name': name,
                                                'path': path})


@deprecated_method
def keyring_file_remove_key(client, name):
    return client.call('keyring_file_remove_key', {'name': name})


@deprecated_method
def keyring_get_keys(client):
    return client.call('keyring_get_keys')


@deprecated_method
def keyring_linux_set_options(client, enable=None):
    params = {}
    if enable is not None:
        params['enable'] = enable
    return client.call('keyring_linux_set_options', params)

File: module/event/subsystems/keyring/keyring.c
/* SPDX-License-Identifier: BSD-3-Clause
 * Copyright (c) 2024 Intel Corporation.  All rights reserved.
 */

#include "spdk/stdinc.h"
#include "spdk/keyring.h"
#include "spdk_internal/init.h"

static void
keyring_subsystem_init(void)
{
	int rc = spdk_keyring_init();

	spdk_subsystem_init_next(rc);
}

static void
keyring_subsystem_fini(void)
{
	spdk_keyring_cleanup();
	spdk_subsystem_fini_next();
}

static void
keyring_subsystem_write_config_json(struct spdk_json_write_ctx *w)
{
	spdk_json_write_array_begin(w);
	spdk_keyring_write_config(w);
	spdk_json_write_array_end(w);
}

static struct spdk_subsystem g_subsystem_keyring = {
	.name = "keyring",
	.init = keyring_subsystem_init,
	.fini = keyring_subsystem_fini,
	.write_config_json = keyring_subsystem_write_config_json,
};

SPDK_SUBSYSTEM_REGISTER(g_subsystem_keyring);
